crates/sui-indexer/src/handlers/event_handler.rs

```diff
@@ -76,26 +79,22 @@ impl EventHandler {
             self.event_handler_metrics
                 .total_events_received
                 .inc_by(event_count as u64);
-            commit_events(&mut pg_pool_conn, event_page.clone())?;
-            // Event page's next cursor can be None when latest event page is reached,
-            // if we use the None cursor to read events, it will start from genesis,
-            // thus here we do not commit / use the None cursor.
-            // This will cause duplicate run of the current batch, but will not cause
-            // duplicate rows b/c of the uniqueness restriction of the table.
-            if let Some(next_cursor_val) = event_page.next_cursor.clone() {
-                commit_event_log(
-                    &mut pg_pool_conn,
-                    Some(next_cursor_val.tx_digest.base58_encode()),
-                    Some(next_cursor_val.event_seq),
-                )?;
+            let commit_result = commit_events(&mut pg_pool_conn, event_page.clone())?;
+
+            if let Some((commit_count, next_cursor_val)) = commit_result {
+                next_cursor = Some(next_cursor_val);
+
+                self.event_handler_metrics.total_event_page_committed.inc();
                 self.event_handler_metrics
                     .total_events_processed
-                    .inc_by(event_count as u64);
-                next_cursor = Some(next_cursor_val);
+                    .inc_by(commit_count as u64);
+                info!(
+                    "Committed {} events, next cursor: {:?}",
+                    commit_count, next_cursor
+                );
             }
-            self.event_handler_metrics.total_event_page_committed.inc();
-            // sleep when the event page has been the latest page
-            if event_count < EVENT_PAGE_SIZE || event_page.next_cursor.is_none() {
+
+            if event_page.next_cursor.is_none() {
```

Comment:
`event_count < EVENT_PAGE_SIZE` indicates `event_page.next_cursor.is_none()`, so no need to check both